src/host/mongomerkle.rs

```diff
@@ -197,27 +126,37 @@ impl<const DEPTH: usize> MongoMerkle<DEPTH> {
     }
 
     pub fn batch_update_records(
-        &self,
+        &mut self,
         records: &Vec<MerkleRecord>,
     ) -> Result<(), mongodb::error::Error> {
-        let dbname = Self::get_db_name();
-        let cname = self.get_collection_name();
-        let collection = db::get_collection::<MerkleRecord>(dbname, cname.clone())?;
-        let (_, new_records) = self.batch_get_records(&records)?;
-        /*
-        println!(
-            "records size is: {:?}, new record size is : {:?}",
-            records.len(),
-            new_records.len()
-        );*/
+        // sort records by index to ensure the parent node is processed before its child nodes.
+        let mut sort_records: Vec<MerkleRecord> = records.clone();
+        sort_records.sort_by(|r1, r2| r1.index.cmp(&r2.index));
+
+        let mut new_records: Vec<MerkleRecord> = vec![];
+        let mut check = true;
+        for record in sort_records {
+            // figure out whether a parent node had been added or not,
+            // to save the cache/db reading for its child nodes for optimizing.
+            if check {
+                match self.get_record(record.index, &record.hash) {
+                    Ok(Some(_)) => {}
+                    _ => {
+                        check = false;
+                        new_records.push(record);
+                    }
+                }
+            } else {
+                new_records.push(record);
+            }
+        }
```

Comment:
This batch_update_records logic changed.
Originally it just try insert the input records to db (if exist, skip. If not exist, insert)
Now the input records must be in one leaf path. (Must be the parend nodes of a leaf)
Currently we only use it for this scenario so if we do this change, better change the name to something like batch_update_records_in_one_leaf_path, etc

Reply:
rename batch_update_records to update_leaf_path_records